### ISSUE TITLE
Heading not generated in `toc` if nested within 2 or more elements (but generated if nested within 1 element)

### ISSUE THREAD

**Issue by ManasMadrecha** [nuxt/content#996]
### Version
@nuxt/content: v1.14.0 
nuxt: v2.15.7

### Steps to reproduce
```md
<!-- This generates Heading → -->

<div>

### Heading 

</div>



<!-- This DOES NOT generate Heading → -->

<div>
<div>

### Heading 

</div>
</div>



<!-- This DOES NOT generate Heading → -->

<details>
<summary>

### Heading 
</summary>

Some content here
</details>
```

### What is Expected?

1. Headings should be generated even if present inside some random Divs or at least if present inside `summary` tag. This is very common use-case while adding FAQs. 

2. Or, at least an option should be given in `nuxt.config.js` in `content` as to the depth/nesting of the elements up to which Headings will be generated even if written in them

**Comment by NozomuIkuta** [comment]
@ManasMadrecha 

Currently, as you described, only root-level or one-deep headings are collected as parts of TOC, and it's hard-corded how deeply Nuxt Content search heading tags.

https://github.com/nuxt/content/blob/ea976cee393a8fef14f5cd7572dfe0dd56473640/packages/content/parsers/markdown/index.js#L47-L50

https://github.com/nuxt/content/blob/ea976cee393a8fef14f5cd7572dfe0dd56473640/packages/content/parsers/markdown/index.js#L32-L40

---

> 1. Headings should be generated even if present inside some random Divs or at least if present inside summary tag. This is very common use-case while adding FAQs.
> 
> 2. Or, at least an option should be given in nuxt.config.js in content as to the depth/nesting of the elements up to which Headings will be generated even if written in them

Whichever is accepted, it brings a new feature to Nuxt Content, which is not actively developed and is supposed to be replaced with Docus. 

So, all we should do will be to notify @pi0 and @Tahul as a reference.

**Comment by NozomuIkuta** [comment]
A little correction:

> Nuxt Content, which is not actively developed and is supposed to be replaced with Docus

Docus doesn't completely replace Nuxt Content, as far as I know.
So what's described in this issue may be in the scope of Nuxt Content v2.

**Comment by ManasMadrecha** [comment]
> Currently, as you described, only root-level or **one-deep** headings are collected as parts of TOC, and it's **hard-corded** how deeply Nuxt Content search heading tags.

@NozomuIkuta So, for the time being, could you kindly change the code to at least 2 level depth? This way at least the headings inside the `<summary>` tag will be generated in TOC 😋

**Comment by NozomuIkuta** [comment]
@ManasMadrecha

Actually, I can't because I'm just a collaborator. All I can do is to submit PRs like other people in community.

Additionally, I believe the maintainer also won't do that instantly because it's basically a breaking change.

Generally, once software reaches to v1, and the more people start to use it, maintainer has to take the more factors into consideration, such as balancing between what to do and what not to, maintainability, performance, and impacts of changes on the existing users and their projects, and so on.

In summary, for the time being, you can utilize [patch-package](https://www.npmjs.com/package/patch-package) to modify the package code **only in your project**.

They may handle your concern in Nuxt Content v2 or Docus since I notified them in the comment above.

I hope this would help.

**Comment by ManasMadrecha** [comment]
@Tahul Amazing. That solves my request.
